network/p2p/p2p.go

```diff
@@ -178,6 +182,12 @@ func MakeService(ctx context.Context, log logging.Logger, cfg config.Local, h ho
 	h.SetStreamHandler(AlgorandWsProtocol, sm.streamHandler)
 	h.SetStreamHandler(libp2phttp.ProtocolIDForMultistreamSelect, sm.streamHandlerHTTP)
 
+	// set an empty handler for telemetryID/telemetryInstance protocol in order to allow other peers to know our telemetryID
+	telemetryID := log.GetTelemetryGUID()
+	telemetryInstance := log.GetInstanceName()
+	telemetryProtoInfo := fmt.Sprintf(algorandGUIDProtocolTemplate, telemetryID, telemetryInstance)
+	h.SetStreamHandler(protocol.ID(telemetryProtoInfo), func(s network.Stream) {})
```

Comment:
This is an interesting way to advertise the telemetry info. I would have expected just `/algorand-telemetry/1.0.0`, with a very basic stream handler that returns the desired info.

Regardless, I think your stream handler should have an `s.Close()`

Reply:
closing is not strictly needed but does not hurt, added.

I considered a basic data exchange but it requires bunch of synchronization code between this and the main stream, imo added complexity isn't worth it.